hack/test_errata.py

```diff
@@ -0,0 +1,598 @@
+import datetime
+import errata
+import github
+import http
+import json
+import os
+import re
+import tempfile
+import unittest
+import urllib
+from collections import namedtuple
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+def test_extract_errata_number_from_body_valid_url():
+    body = "errata is https://errata.devel.redhat.com/advisory/12345"
+    assert errata.extract_errata_number_from_body(body) == 12345
+
+    body = "errata is https://errata.devel.redhat.com/advisory/9999"
+    assert errata.extract_errata_number_from_body(body) == 9999
+
+    body = "errata is https://errata.devel.redhat.com/advisory/"
+    assert errata.extract_errata_number_from_body(body) == None
+
+    body = "errata is https://errata.devel.redhat.com/advisory/invalid"
+    assert errata.extract_errata_number_from_body(body) == None
+
+
+def test_extract_errata_number_from_body_invalid_url():
+    # Possible invalid body
+    # body = "errata is https://errata.devel.redhat.com/advisory/advisory/12345"
+    # assert errata.extract_errata_number_from_body(body) == None
+
+    body = "errata is https://errata.devel.redhat.com/"
+    assert errata.extract_errata_number_from_body(body) == None
+
+
+def test_load():
+    with tempfile.TemporaryDirectory() as tempdir:
+        cachepath = os.path.join(tempdir, "cache.json")
+        cache = {"foo": "bar"}
+
+        with open(cachepath, 'w') as file:
+            json.dump(cache, file)
+
+        loaded_cache = errata.load(cachepath)
+        assert loaded_cache == cache
+
+
+def test_load_nonexisting_file():
+    with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
[tmpdir fixture](https://docs.pytest.org/en/stable/tmpdir.html)

Reply:
[`TemporaryDirectory` docs][1]:

> On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.

So this should be fine as it stands without involving pytest-specific stuff.  But if we stick with pytest as the overall framework, using their tempdir fixture is probably more idiomatic than using equivalent stdlib tooling.

[1]: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory